### ISSUE TITLE
API key invalid or rate limited

### ISSUE THREAD

**Issue by Dalk21** [qconrad/intellistar-emulator#53]
The TWC API key used for fetching weather info is currently not working,

If it's fully broken, you can actually inspect API calls from both weather.com and wunderground.com to get valid API keys (not sure on how long they work)

**Comment by Dalk21** [comment]
A user in issue #52 shows the issue

**Comment by TerryThor731** [comment]
I hope I can find this helpful.

**Comment by TerryThor731** [comment]
I hope we can get this resolved ASAP.

**Comment by Halloween20** [comment]
and the narrations too

**Comment by TerryThor731** [comment]
> and the narrations too

Narrations, really?

**Comment by Halloween20** [comment]
Yes

**Comment by Dalk21** [comment]
If you need I can get a fork setup soon with a working API key

**Comment by TerryThor731** [comment]
> If you need I can get a fork setup soon with a working API key

You can try to do that. If you please.

**Comment by Halloween20** [comment]
Thank you so much

**Comment by Halloween20** [comment]
when do you think this emulator could be working Again By ?

**Comment by TerryThor731** [comment]
I don't know exactly when it will be resolved, but once it is, everything should start working properly again. For now, please be patient while they work on fixing the issue.

**Comment by Dalk21** [comment]
https://dalk21.github.io/intellistar-emulator/

I made a fork and put a working API key on there

Don't expect it to work forever, but it should work for the next good while

**Comment by Halloween20** [comment]
are they going to have make a new weather radar ?

**Comment by K-A-0-S** [comment]
The api key is broke... 2 minute I fixed.

**Comment by Halloween20** [comment]
I Notice the weather radar is lagging is there enyway some one could fix that problem if you can that would be great and thank you

**Comment by K-A-0-S** [comment]
The api key needs to be replaced. That will fix it. But it seems the author has lost interest. Mine works, Simple fix.

**Comment by Dalk21** [comment]
I've already set mine up with a working key lol, no need. I'll look into getting the TWC radar on there. Should be simple

**Comment by K-A-0-S** [comment]
The solution is easy. Replace the API Key. Problem solved.

**Comment by Dalk21** [comment]
For radar, I already have a fix and can work on implementing it here soon. It uses the weather channel radar + mapbox to show the radar for the location.

**Comment by K-A-0-S** [comment]
> I Notice the weather radar is lagging is there enyway some one could fix that problem if you can that would be great and thank you

Yes... the radar lags. Unless you are using something other than a half way decent desktop computer. It never even fully loads on my fire tv. This is because of the high rez image and 20 frames being loaded.

Anyway.. there is a way around this.

**Comment by Dalk21** [comment]
If you want, I'll send over the radar docs and apis for you to mess with, feel free to make a pull request with those

**Comment by K-A-0-S** [comment]
I have it working with NWS low rez radar..

![image](https://github.com/qconrad/intellistar-emulator/assets/169946395/ea0b2b8a-9e52-4c9c-8d90-c877cf3b1f36)

**Comment by K-A-0-S** [comment]
Also Local Radar...

![image](https://github.com/qconrad/intellistar-emulator/assets/169946395/3cbeaf67-e0d2-44e8-8a45-66e89989685a)

**Comment by K-A-0-S** [comment]
> If you want, I'll send over the radar docs and apis for you to mess with, feel free to make a pull request with those

I am interested.

**Comment by Halloween20** [comment]
Please do that

**Comment by Halloween20** [comment]
If somebody can add the voice to the forecast that would be great thanks

**Comment by Dalk21** [comment]
> > If you want, I'll send over the radar docs and apis for you to mess with, feel free to make a pull request with those
> 
> I am interested.

Where would you like me to send them to

**Comment by K-A-0-S** [comment]
Can you put them in your repository?

**Comment by Dalk21** [comment]
I put it there, it's the IBM public ones, find weather imagery and use the key in my repo for it

**Comment by Halloween20** [comment]
is the local on the 8s going to have a new look ?

**Comment by KiernanTheFemboy** [comment]
> For radar, I already have a fix and can work on implementing it here soon. It uses the weather channel radar + mapbox to show the radar for the location.

Haii, Can you show me how to do this? I hate the current radar on the simulator. I have discord and insta and stuff if you want to talk about it privately.

**Comment by Dalk21** [comment]
I can get it worked on soon. I'll make a pull request and hopefully we can get it publicly available

**Comment by test-240123** [comment]
I'm Dalk (another account), but I've worked on this on my repo, and have got TWC radar on there. Radar needs to be worked on to possibly preload images to make them not lag. But other than that, it works.

**Comment by Halloween20** [comment]
do know how long is going to be ?

**Comment by Halloween20** [comment]
if you give me a time on it that would help Me A Lot

**Comment by KiernanTheFemboy** [comment]
> I'm Dalk (another account), but I've worked on this on my repo, and have got TWC radar on there. Radar needs to be worked on to possibly preload images to make them not lag. But other than that, it works.

Hey I decided to try it out and the radar doesn't work on the zoomed radar slide. if possible can you get that working too? and zoom it out on the regular radar slide.

**Comment by K-A-0-S** [comment]
I am KAOS and have already figured all of this out.

**Comment by Halloween20** [comment]
how long will be until this emulator starts working again ?

**Comment by K-A-0-S** [comment]
> how long will be until this emulator starts working again ?

It works here...

https://dalk21.github.io/intellistar-emulator/

and here...

https://k-a-0-s.github.io/

I also have a low rez radar version...

![image](https://github.com/qconrad/intellistar-emulator/assets/169946395/8904765b-c53c-4e69-9b66-a425f277f640)

That radar loads immediately... no lag.

In order for the hi-rez radar version to work without lag on low hardware resource devices, it needs to be pre-loaded.

As for narration... that can be addressed after other issues are resolved.

**Comment by Dalk21** [comment]
You should be able to modify the radar code on https://github.com/test-240123/intellistar-emulator and make it preload assets

**Comment by K-A-0-S** [comment]
> You should be able to modify the radar code on https://github.com/test-240123/intellistar-emulator and make it preload assets

Not as easy as you make it sound. Otherwise, it would have been done. But it is not... there is no additional functionality beyond the original code other than fixing the api key... a simple fix I made long before these comments. The original developer seems to have abandoned this project... or perhaps is busy. Regardless... Most people are not going to be able to "fix" the radar issue with this type of limited information... "You should be able to modify the radar code on https://github.com/test-240123/intellistar-emulator and make it preload assets"

I will leave it at that.

**Comment by K-A-0-S** [comment]
![image](https://github.com/qconrad/intellistar-emulator/assets/169946395/72169cf4-118e-4e4e-abc9-c38f6e4a6ad5)

**Comment by Halloween20** [comment]
i been checking for updates on this intellistar emulator

**Comment by K-A-0-S** [comment]
I fixed the radar lag. Fix is here...

https://k-a-0-s.github.io/

Enjoy!

**Comment by Nacroni** [comment]
> I Notice the weather radar is lagging is there enyway some one could fix that problem if you can that would be great and thank you

the radar is powered by NWS using a webpage (i think it's an iframe), it's gonna lag since it needs to load everything.

**Comment by Halloween20** [comment]
when do you think can you put in the Narrations in ?
